crates/holochain/src/conductor/api/api_external/admin_interface.rs

```diff
@@ -219,80 +220,237 @@ impl InterfaceApi for RealAdminInterfaceApi {
+        /// The `CellId` for which to dump state
         cell_id: Box<CellId>,
     },
 }
 
-/// Responses to messages received on an Admin interface
+/// Represents the possible responses to an [`AdminRequest`]
+/// and follows a general convention of `noun_verb` as opposed to
+/// the `verb_noun` of `AdminRequest`.
+///
+/// Will serialize as an object with any contents of the enum on a key `data`
+/// and the enum variant on a key `type`, e.g.
+/// `{ type: 'app_interface_attached', data: { port: 4000 } }`
+///
+/// [`AdminRequest`]: enum.AdminRequest.html
 #[derive(Debug, serde::Serialize, serde::Deserialize, SerializedBytes)]
 #[cfg_attr(test, derive(Clone))]
 #[serde(rename_all = "snake_case", tag = "type", content = "data")]
 pub enum AdminResponse {
-    /// This response is unimplemented
-    Unimplemented(AdminRequest),
-    /// hApp [Dna]s have successfully been installed
+    /// Can occur in response to any [`AdminRequest`].
+    ///
+    /// There has been an error during the handling of the request.
+    /// See [`ExternalApiWireError`] for variants.
+    ///
+    /// [`AdminRequest`]: enum.AdminRequest.html
+    /// [`ExternalApiWireError`]: error/enum.ExternalApiWireError.html
+    Error(ExternalApiWireError),
+    /// The succesful response to an [`AdminRequest::InstallApp`].
+    ///
+    /// The resulting [`InstalledApp`] contains the App id,
+    /// the [`CellNick`]s and, most usefully, the new [`CellId`]s
+    /// of the newly installed `Dna`s. See the [`InstalledApp`] docs for details.
+    ///
+    /// [`AdminRequest::InstallApp`]: enum.AdminRequest.html#variant.InstallApp
+    /// [`InstalledApp`]: ../../../holochain_types/app/struct.InstalledApp.html
+    /// [`CellNick`]: ../../../holochain_types/app/type.CellNick.html
+    /// [`CellId`]: ../../../holochain_types/cell/struct.CellId.html
     AppInstalled(InstalledApp),
-    /// AdminInterfaces have successfully been added
-    AdminInterfacesAdded(()),
-    /// A list of all installed [Dna]s
-    ListDnas(Vec<DnaHash>),
-    /// Keystore generated a new AgentPubKey
-    GenerateAgentPubKey(AgentPubKey),
-    /// Listing all the cell ids in the conductor
-    ListCellIds(Vec<CellId>),
-    /// Listing all the active app ids in the conductor
-    ListActiveAppIds(Vec<AppId>),
-    /// [AppInterfaceApi] successfully attached
+    /// The succesful response to an [`AdminRequest::AddAdminInterfaces`].
+    ///
+    /// It means the `AdminInterface`s have successfully been added
+    ///
+    /// [`AdminRequest::AddAdminInterfaces`]: enum.AdminRequest.html#variant.AddAdminInterfaces
+    AdminInterfacesAdded,
+    /// The succesful response to an [`AdminRequest::GenerateAgentPubKey`].
+    ///
+    /// Contains a new `AgentPubKey` generated by the Keystore
+    ///
+    /// [`AdminRequest::GenerateAgentPubKey`]: enum.AdminRequest.html#variant.GenerateAgentPubKey
+    AgentPubKeyGenerated(AgentPubKey),
+    /// The succesful response to an [`AdminRequest::ListDnas`].
```

Comment:
`s/succesful/successful/g`

```suggestion
    /// The successful response to an [`AdminRequest::ListDnas`].
```